notebooks/patch_to_point/patch_to_point.ipynb

```diff
@@ -0,0 +1,689 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Patch to point extraction pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As an input the script should take the path of the STAC collection for Sentinel-1 and Sentinel-2 patch extractions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As a first step, the S1 and S2 STAC collections have to be split per EPSG, as openeo's `load_stac` can only handle one CRS in a STAC collection.\n",
+    "This can be accomplished with GFMap's `split_collection_by_epsg`, as shown in the code block below. \n",
+    "\n",
+    "NOTE: In the finale pipeline the STAC collections will be on /vitodata/worldcereal_data. For example purposes, my own public folder is used."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from openeo_gfmap.utils import split_collection_by_epsg\n",
+    "\n",
+    "# stac_path = '/data/users/Public/vincent.verelst/world_cereal/s1_s2_meteo_extraction/Sentinel2/stac/collection.json'\n",
+    "# output_dir = '/data/users/Public/vincent.verelst/world_cereal/s1_s2_meteo_extraction/Sentinel2/small-split-stac/'\n",
+    "\n",
+    "# split_collection_by_epsg(stac_path, output_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, the script will generate a list of all split STAC collections. As it isn't know up front how many different CRS were in the original STAC collection, it's not known into how many STAC collections it will be split.\n",
+    "\n",
+    "An assertion is added to make sure that the number of S1 and S2 STAC collections is the same. (As normally both S1 and S2 STAC collection should have the same spatiotemporal extent.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a list of the paths to the split collections\n",
+    "import glob\n",
+    "s1_collection_paths = glob.glob(\"/data/users/Public/vincent.verelst/world_cereal/s1_s2_meteo_extraction/Sentinel1/split_stac/*/collection.json\")\n",
+    "s2_collection_paths = glob.glob(\"/data/users/Public/vincent.verelst/world_cereal/s1_s2_meteo_extraction/Sentinel2/small-split-stac/*/collection.json\")\n",
+    "\n",
+    "assert len(s1_collection_paths) == len(s2_collection_paths), \"Mismatch in number of collections\""
```

Comment:
we can imagine that often these collections are not fully in sync. One may still be extracting, so this assertion could fail. Wouldn't a better option be to somehow make an intersect of the two collections and only keep those that are "identical" for both inputs?

Reply:
This would make it difficult to track which 'part' of the STAC collection has been already extracted to points and which not. Would it not be safer to wait until all S1 and S2 patch extractions are finished for a certain dataset before we trigger the point extractions?

Reply:
do they all come with their own collections? I think i don't get the entire picture on collection organization yet and how the growing database of extractions will be handled. Yes we should wait till the extractions for "a dataset" are done, but aren't there other ways as well to check if points are already extracted from patches? Looking into the existing geoparquet e.g.? let's brainstorm a bit about this on monday.